Add `sentry-dart-plugin` documentation (#7097)
getsentry/sentry-docs: src/platforms/flutter/troubleshooting.mdx

## Source Context

Source Context support requires compiling your app using the `split-debug-info` build parameter and the latest Flutter (`master` channel), you must also upload [debug symbols](/platforms/flutter/upload-debug/).

## Sentry Dart Plugin

> If you are using [Sentry Dart Plugin](/platforms/flutter/upload-debug/#automatically-Upload-debug-symbols) to upload debug symbols, refer to the points below to resolve potential issues.

Reviewer: ```suggestion
If you are using the Sentry Dart Plugin to upload [Debug Symbols](/platforms/flutter/upload-debug/#automatically-Upload-debug-symbols), refer to the points below to resolve potential issues.
```

> Sentry's `auth_token` requires the `project:releases` or `project:write` scope, See [docs](/product/cli/dif/#permissions).

Reviewer: ```suggestion
Sentry's `auth_token` requires the `project:releases` or `project:write` scope. See [Debug Information Files](/product/cli/dif/#permissions) docs for more information.
```

> For the `commits` feature, Sentry's `auth_token` also requires the `org:read` scope, See [docs](/api/permissions/#releases).

Reviewer: ```suggestion
For the `commits` feature, Sentry's `auth_token` also requires the `org:read` scope. See API [Permissions & Scopes](/api/permissions/#releases) docs for more information.
```

> Dart's `--obfuscate` option is required to be paired with `--split-debug-info` to generate a symbol map, See [docs](https://github.com/flutter/flutter/wiki/Obfuscating-Dart-Code).

Reviewer: ```suggestion
Dart's `--obfuscate` option is required to be paired with `--split-debug-info` to generate a symbol map. See [Dart docs](https://github.com/flutter/flutter/wiki/Obfuscating-Dart-Code) for more information.
```

> The `--split-debug-info` option requires setting a output directory, the directory must be an inner folder of the project's folder, See [docs](https://docs.flutter.dev/deployment/obfuscate#obfuscate-your-app).

Reviewer: ```suggestion
The `--split-debug-info` option requires setting an output directory. The directory must be an inner folder of the project's folder. See [Flutter docs](https://docs.flutter.dev/deployment/obfuscate#obfuscate-your-app) for more information.
```

> Flutter's `build web` command requires setting the `--source-maps` parameter to generate source maps, See [Issue](https://github.com/flutter/flutter/issues/72150#issuecomment-755541599)

Reviewer: ```suggestion
Flutter's `build web` command requires setting the `--source-maps` parameter to generate source maps. See [Flutter GitHub Issue](https://github.com/flutter/flutter/issues/72150#issuecomment-755541599) for more information.
```
